Microsoft.Toolkit.Uwp.UI.Controls/ImageEx/ImageExBase.cs

```diff
@@ -225,15 +241,36 @@ private void OnImageFailed(object sender, ExceptionRoutedEventArgs e)
             VisualStateManager.GoToState(this, FailedState, true);
         }
 
-        private void ImageExBase_EffectiveViewportChanged(FrameworkElement sender, EffectiveViewportChangedEventArgs args)
+        private void ImageExBase_LayoutUpdated(object sender, object e)
+        {
+            InvalidateLazyLoading();
+        }
+
+        private void InvalidateLazyLoading()
         {
-            var bringIntoViewDistanceX = args.BringIntoViewDistanceX;
-            var bringIntoViewDistanceY = args.BringIntoViewDistanceY;
+            if (!IsLoaded)
+            {
+                _isInViewport = false;
+                return;
+            }
+
+            var hostElement = GetHostElement();
```

Comment:
@h82258652 you can use the visual tree helpers for this vs. a custom method: i.e. `this.FindAscendant<ScrollViewer>()` see [docs here](https://docs.microsoft.com/en-us/windows/communitytoolkit/extensions/visualtree)